push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/SenderServiceImpl.java

```diff
@@ -67,6 +67,8 @@ public void send(PushApplication pushApplication, UnifiedPushMessage message) {
                         message.getClientIdentifier()
                         );
 
+        //Let's put a unique identifier into the payload to be able to do some analytics around that Push Notification
+        message.getMessage().getUserData().put("push-identifier",pushMessageInformation.getId());
```

Comment:
Don't we want to keep the user data for the 'user', I think it's better to put it in another location